Formulaire de consultation sur 2 columne + fiche patient plus élégante

JocelynDelalande opened the pull request:
Résoud la demande #83 
Concerne #20 

TODO restante : 



- [x] un·e utilisateur·ice non admin ne peut pas sauvegarder une fiche consultation fraichement créée (manque les boutons édition/fin édition)
- [x]  Parfois le texte "déborde" de la case du champ de formulaire (ex avec le champ "Loisirs", à priori c'est le seul constaté pour l'instant)
- [x] vérifier que l'apparition des boutons supprimer fonctionne

JocelynDelalande committed:
Re-implement tab change detection with events

I Believe this is clearer and cleaner than using MutationObserver. It also
makes it more generic (pave the way for handling router-view change the same way).

ref #70

JocelynDelalande committed:
Auto-save on tab change only if form changed

Saves some requests.

JocelynDelalande committed:
Only handle relevant tab change event

Only if it is related to current form tab.

Using `:visible` is quite a hack. Properly handling that would require that the
tab itself $broadcast from its scope (narrowing the receivers list). Currently,
its `PatientCtrl` who issues $broadcast.

JocelynDelalande committed:
Refactor unsaved tab/form code and allow to prompt user

Make it more generic, pave the way for #87

JocelynDelalande committed:
Ask confirmation to user when he leaves without saving examination data

fix #87

JocelynDelalande committed:
Ask confirmation before leaving page with unsaved data

As commented in the patch, we can do little on browser page quit (no autosave),
so this is barely a confirmation message.

fix #115

JocelynDelalande committed:
Avoid repeating closing event handling for unsaved stuff

- avoid handling several $locationChangeStart on same click.
- avoir handling $locationChangeStart and uiTabChange on same click.

JocelynDelalande committed:
Make EditFormManager capable of actioning several forms

At once, all visible forms.

Also refactored slightly and documented.

JocelynDelalande committed:
Refactor laterality key/values with service + filter

To make it more reusable outside of PatientCtrl

JocelynDelalande committed:
Refactor sex key/values with service + filter

To make it more reusable outside of PatientCtrl and consistent with laterality handling.

JocelynDelalande committed:
Fix bug preventing to add doctors

Not mentioning meta.fields is forbidden with recent DRF versions.

JocelynDelalande committed:
Refactor doctor selector into a dedicated component

Pave the way to display examination in both examination and patient details
view.

JocelynDelalande committed:
Do not leave edit mode after adding a new doctor

fix #132

JocelynDelalande committed:
Comments cleanup

iso-functional commit

JocelynDelalande committed:
Refactor spheres handling in examination controller

To avoid repetition

iso-functional commit.

JocelynDelalande committed:
Remove dead code

iso-functional commit

JocelynDelalande committed:
Move isEmpty() to utils

iso-functional commit

JocelynDelalande committed:
Remove debug logging

iso-functional commit

JocelynDelalande committed:
Switch to 2 columns examination view layout

fix #83

JocelynDelalande committed:
Display patient age near patient name

in examination/patient view

JocelynDelalande committed:
Re-organize left pane of patient view

- align items in two columns
- add a thin border between the two cols
- field title and field input are now inline (to avoid weird space between name
  and input)
- say « not documented » for most field, when relevant (instead of white space)

JocelynDelalande committed:
Style undocumented values a more light maneer

When the information is not filled, it seems more logical to dim the info
message saying  it rather than highlighting it.

JocelynDelalande committed:
Add propper spacing to patient notes

(before, text was glued to panel borders)

JocelynDelalande committed:
Use less agressive color for examination reason

JocelynDelalande committed:
Integrate examination type to first panel heading, as badge

Save some vertical space and makes it acceptable for people not using this
information.

JocelynDelalande committed:
Refactor optional module to allow  configuring several views

And not only dashboard

JocelynDelalande committed:
Add forgotten i18n string

JocelynDelalande committed:
Allow disabling spheres in examination view

JocelynDelalande committed:
Fix false-positive unsaved data confirmation dialog

angular-xeditable offers no guarantee at handling correctly the dirty <form>
status.

Thus, now rely only the dirty status on individual fields to trigger or not the
unsaved data user confirmation dialog.

ref #113 #114

jbgury commented on libreosteoweb/static/js/app/editformmanager.js:
```diff
@@ -17,14 +17,20 @@
 */
 var editFormManager = angular.module('loEditFormManager', []);
 
-var MutationObserver = window.MutationObserver || window.WebKitMutationObserver || window.MozMutationObserver;
```
Le retrait de ceci risque de perturber l'affichage correct des boutons du menu en haut lors des changements d'onglet. C'est un risque à confirmer...

JocelynDelalande replied:
Non, j'ai vérifié, à part sur de très vieilles versions de navigateur (genre Firefox 15), ça ne va rien changer.

JocelynDelalande committed:
Clarify separation between consultation reason and medical examination

JocelynDelalande committed:
Fix bad modal button displaying/hiding

My code to handle multiple forms via the modal buttons was pretty much
buggy/unfinished. Now, it is better and works as expected :-)

JocelynDelalande committed:
Also auto-save examination tab

drawback is that it close the form and requires more clicks.

JocelynDelalande committed:
Fix non-admin user rights on /profiles/get_by_user

Non-admin users were no longer able to (at least) :

- Fetch their profile
- Create new examination
- Create new patient

That is unrelated to current PR (bug also existed in current develop branch)

JocelynDelalande committed:
Fix Patient.hobbies input overflow in examination view

JocelynDelalande committed:
Add missing migration merge

JocelynDelalande committed:
Cleanup examination edit view

- rename editableForm → examinationForm to be less confusing
- group edit/save behaviours of the two forms of the view instead of handling
  them half-appart.

JocelynDelalande committed:
Fix action buttons display on new examination

They were not appearing, and different parts of the code were
attempting (unsuccessfuly) to make it happen.

Fixed the bug and cleaned the bug !

JocelynDelalande committed:
Fix Missing e-names in patient detail form

JocelynDelalande committed:
Fix typo in var name

JocelynDelalande committed:
Refactor EditFormManager Through FormAction

Clarifies what this part of the code does.

JocelynDelalande committed:
Fix selenium tests
